xfer/multi_client.go

```diff
@@ -0,0 +1,104 @@
+package xfer
+
+import (
+	"log"
+	"sync"
+	"time"
+)
+
+// ProbeConfig ...
+type ProbeConfig struct {
+	Token    string
+	ProbeID  string
+	Insecure bool
+}
+
+type multiClient struct {
+	handler ControlHandler
+	ProbeConfig
+
+	mtx     sync.Mutex
+	sema    semaphore
+	clients map[string]AppClient
+	quit    chan struct{}
+}
+
+// MultiAppClient ...
+type MultiAppClient interface {
+	Set(hostname string, endpoints []string)
+}
+
+// NewMultiAppClient ...
+func NewMultiAppClient(pc ProbeConfig, handler ControlHandler) MultiAppClient {
+	return &multiClient{
+		ProbeConfig: pc,
+		handler:     handler,
+
+		sema:    newSemaphore(maxConcurrentGET),
+		clients: map[string]AppClient{},
+		quit:    make(chan struct{}),
+	}
+}
+
+// Set ...
+func (c *multiClient) Set(hostname string, endpoints []string) {
+	for _, endpoint := range endpoints {
+		go func(endpoint string) {
+			c.sema.p()
+			defer c.sema.v()
```

Comment:
Yup.  They were not introduced in this change, so I suggest we file a techdebt ticket and fix them up lated.